packages/@aws-cdk/aws-appsync/lib/mapping-template.ts

```diff
@@ -40,6 +40,26 @@ export abstract class MappingTemplate {
     return this.fromString('{"version" : "2017-02-28", "operation" : "Scan"}');
   }
 
+  /**
+   * Mapping template to list a set of items from a DynamoDB table
+   * 
+   * @param defaultLimit the default value for the limit argument
+   * @param limitKeyName the name of the limit argument
+   * @param nextTokenKeyName the name of the next token argument
+   * 
+   * @default 
+   * defaultLimit = 100
+   * limitKeyName = 'limit'
+   * nextTokenKeyName = 'nextToken'
+   */
+  public static dynamoDbListItems(
+    defaultLimit?: number, 
+    limitKeyName?: string,
+    nextTokenKeyName?: string
+  ): MappingTemplate {
+    return this.fromString(`{"version" : "2017-02-28", "operation" : "Scan", "limit": $util.defaultIfNull($ctx.args.${limitKeyName || 'limit'}, ${defaultLimit || 100}), "nextToken": $util.toJson($util.defaultIfNullOrBlank($ctx.args.${nextTokenKeyName || 'nextToken'}, null))}`);
```

Comment:
[Docs](https://docs.amazonaws.cn/en_us/appsync/latest/devguide/resolver-mapping-template-reference-dynamodb.html#aws-appsync-resolver-mapping-template-reference-dynamodb-scan) indicate that it is obtained from a previous query, so why do we specify a default for `nextToken` here?

`limit` is also optional. Why do we provide a default of 100 here? Is this a hidden maximum set by appsync?

Reply:
Coming from Amplify, the default keynames were "nextToken", "limit", and the default limit value was 100, however I believe this can be removed to follow the doc formats.

edit: yes, btw appsync limits at 100 unless you request a limit increase.